### PR TITLE
feat(ras-acc): keyboard navigation improvements

### PR DESCRIPTION
**Pull request by laurelfulford**
### All Submissions:

* [ ] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR, with https://github.com/Automattic/newspack-plugin/pull/3164, adds some accessibility improvements to the reader auth and modal checkout. Both PRs should be tested together, and the testing steps can be found [here](https://github.com/Automattic/newspack-plugin/pull/3164).

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/modal-checkout/modal.js [hunk 1]
```diff
@@ -27,6 +28,7 @@ domReady( () => {
 	const modalContent = modalCheckout.querySelector( `.${ MODAL_CLASS_PREFIX }__content` );
 	const modalCheckoutHiddenInput = createHiddenInput( 'modal_checkout', '1' );
 	const spinner = modalContent.querySelector( `.${ CLASS_PREFIX }__spinner` );
+	let modalTrigger = document.querySelector( '.newspack-reader__account-link' )[0];
```
Comment (chickenn00dle) [inline src/modal-checkout/modal.js]:
```suggestion
	let modalTrigger = document.querySelector( '.newspack-reader__account-link' )?.[0];
```

**Review by chickenn00dle — CHANGES_REQUESTED** [review]
Left my main code review in the plugin PR, but also had a bit of feedback here.

For the `trapFocus` function, rather than just duplicating the code across the repos, I wonder if we can do something like assign it to browser's global object in plugin, and call it here that way?

**Comment by laurelfulford** [comment]
> For the `trapFocus` function, rather than just duplicating the code across the repos, I wonder if we can do something like assign it to browser's global object in plugin, and call it here that way?

This is a good point! My concern was that the version here in the Newspack Blocks gets super specific in the `is ( iframe )` check, to work with elements that get hidden inside of the modal checkout, and iframes. The version in Newspack Plugin is more of a non-specific, reusable version of the code. 

The `iframe`-specific version is only used in one spot; the other two in Newspack Blocks would use the more generic version. Would it make sense to rework the `trapFocus` in the Newspack Block to cater to just the modal checkout, and to call the generic version from the Newspack Plugin? 

(Just let me know if I'm not making sense!)

File: src/modal-checkout/modal.js (hunk 1 above)
Comment (laurelfulford) [reply src/modal-checkout/modal.js]: Thanks @chickenn00dle! This has been fixed in [fc3cee3](https://github.com/Automattic/newspack-blocks/pull/1760/commits/fc3cee36fcd5a9ce315c5c56016c95bb083d2dd4).

**Review by chickenn00dle — APPROVED** [review]
Thanks @laurelfulford!

> My concern was that the version here in the Newspack Blocks gets super specific in the is ( iframe ) check, to work with elements that get hidden inside of the modal checkout, and iframes. The version in Newspack Plugin is more of a non-specific, reusable version of the code.

Ah. I see what you mean. Makes sense.

> The iframe-specific version is only used in one spot; the other two in Newspack Blocks would use the more generic version. Would it make sense to rework the trapFocus in the Newspack Block to cater to just the modal checkout, and to call the generic version from the Newspack Plugin?

I'm also just remembering its possible for modal checkout to be on a site without the main newspack plugin present (although from what I understand this isn't really a thing?). Given this is a possibility, I think I take back my previous comment and am okay with the duplicate code here. Sorry for the confusion.

**Review by chickenn00dle — APPROVED** [review]

**Comment by laurelfulford** [comment]
Thanks @chickenn00dle!

**Pull request merged**